bpl/neutral_dixon_coles.py

```diff
@@ -0,0 +1,492 @@
+        away_team: jnp.array,
+        num_teams: int,
+        home_goals: Iterable[int],
+        away_goals: Iterable[int],
+        neutral_venue: Iterable[int],
+        team_covariates: Optional[np.array],
+    ):
+        std_home_advantage = numpyro.sample(
+            "std_home_advantage", dist.HalfNormal(scale=1.0)
+        )
+        std_attack = numpyro.sample("std_attack", dist.HalfNormal(scale=1.0))
+        std_defence = numpyro.sample("std_defence", dist.HalfNormal(scale=1.0))
+        mean_defence = numpyro.sample("mean_defence", dist.Normal(loc=0.0, scale=1.0))
+        corr_coef = numpyro.sample("corr_coef", dist.Normal(0.0, 1.0))
+
+        mean_home_advantage = numpyro.sample(
+            "mean_home_advantage", dist.Normal(0.1, 0.2)
+        )
+
+        u = numpyro.sample("u", dist.Beta(concentration1=2.0, concentration0=4.0))
+        rho = numpyro.deterministic("rho", 2.0 * u - 1.0)
+
+        if team_covariates is not None:
+            standardised_covariates = (
+                team_covariates - team_covariates.mean(axis=0)
+            ) / team_covariates.std(axis=0)
+            num_covariates = standardised_covariates.shape[1]
+
+            with numpyro.plate("covariates", num_covariates):
+                attack_coefficients = numpyro.sample(
+                    "attack_coefficients", dist.Normal(loc=0.0, scale=1.0)
+                )
+                defence_coefficients = numpyro.sample(
+                    "defence_coefficients", dist.Normal(loc=0.0, scale=1.0)
+                )
+
+            attack_prior_mean = jnp.matmul(
+                standardised_covariates, attack_coefficients[..., None]
+            ).squeeze(-1)
+            defence_prior_mean = mean_defence + jnp.matmul(
+                standardised_covariates, defence_coefficients[..., None]
+            ).squeeze(-1)
+        else:
+            attack_prior_mean = 0.0
+            defence_prior_mean = mean_defence
+
+        with numpyro.plate("teams", num_teams):
+            standardised_attack = numpyro.sample(
+                "standardised_attack", dist.Normal(loc=0.0, scale=1.0)
+            )
+            standardised_defence = numpyro.sample(
+                "standardised_defence",
+                dist.Normal(
+                    loc=rho * standardised_attack, scale=jnp.sqrt(1.0 - rho**2.0)
+                ),
+            )
+
+            with reparam(config={"home_attack_advantage": LocScaleReparam(centered=0)}):
+                home_attack_advantage = numpyro.sample(
+                    "home_attack_advantage",
+                    dist.Normal(mean_home_advantage, std_home_advantage),
+                )
+            with reparam(
+                config={"away_attack_disadvantage": LocScaleReparam(centered=0)}
+            ):
+                away_attack_disadvantage = numpyro.sample(
+                    "away_attack_disadvantage",
+                    dist.Normal(mean_home_advantage, std_home_advantage),
```

Comment:
should have their own parameters for the mean and st-dev (i.e `mean_away_attack_disadv` and `std_away_attack_disadv`) and add corresponding priors for them above

I would actually maybe suggest dropping the `_advantage` and `_disadvantage` and then just have the attack and defence variables interpreted like we have already (so a positive attack means good attack, and positive defence means good attack - and make it *minus* the defence parameter)

Reply:
I see - this makes sense. I'm not a good enough Bayesian to argue otherwise but I just felt that they needed their own, even if they are the same prior. Maybe just a clean up of the variable names so that it's clear they're meant to be a general prior for these parameters would be good

Reply:
To be clear I'm asking whether that makes sense because I have no idea really 😄

Reply:
Just to update this thread - I now agree that we need a separate variable for all of these as we're taking a sample from the prior, rather than defining it. Ryan's done this in #14